how-to-guides/re-run-failed-training/notebooks/re_run_failed_training.ipynb

```diff
@@ -0,0 +1,489 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "THIzFFk81nme"
+   },
+   "source": [
+    "# How to re-run failed training\n",
```

Comment:
One cool thing that you can do to improve notebooks readability (but this applies only to notebooks) is to add imports to the cell where you actually need the imported lib.

For example neptune import go to code cell in step 1, all torch imports go to cell in step 5.

In this way, you introduce these imports in the context where they are actually used. This improves readability as you have more self-contained code presentation. I saw few guides built this way and it works really well, no need to scroll up to see what is actually being used.

Again, this applied only to notebooks as they have special structure

Reply:
Another advantage is that you do not have this code imports cell at the top

Reply:
True!
I will make the change